client/db.go

```diff
@@ -226,8 +238,12 @@ type Store interface {
 	// GetAllIndexes returns all the indexes that currently exist within this [Store].
 	GetAllIndexes(context.Context) (map[CollectionName][]IndexDescription, error)
 
-	// ExecRequest executes the given GQL request against the [Store].
-	ExecRequest(context.Context, string) *RequestResult
+	// ExecRequest executes the given GQL request against the [Store], with the given identity.
+	ExecRequest(
+		ctx context.Context,
+		identity immutable.Option[string],
```

Comment:
thought: This new param does dirty the interface a little bit IMO, and will be needed for most functions not just this.  I'm fine with this being out of scope for now, but in a follow up I would like to see something like what we did for transactions - with a `WithIndentity() DB` func on the `DB` interface.

I think this should be done quite soon, as currently all the collection funcs seem to bypass ACP, meaning ACP can be bypassed for unrestricted document access via the CLI and Http clients.

Reply:
There is something like that already planned for with Auth Tokens will be implemented, John had a suggestion to inject it into the context even.

- [x] Made an issue ([Cleanup The Identity Param](https://github.com/sourcenetwork/defradb/issues/2439)) will do out of this PR: